proto/sentry_protos/sentry/v1/taskworker.proto

```diff
@@ -91,3 +92,52 @@ message InflightActivation {
   // If processing is not complete by this time, a retry will be attempted.
   optional google.protobuf.Timestamp processing_deadline = 6;
 }
+
+////////////////////////////
+// RPC messages and services
+////////////////////////////
+message Error {
+  // Taken directly from the grpc docs.
+  int32 code = 1;
+  string message = 2;
+
+  // A list of messages that carry any error details.
+  repeated google.protobuf.Any details = 3;
+}
+
+service ConsumerService {
+  // Fetch a new task activation to process.
+  rpc GetTask(GetTaskRequest) returns (GetTaskResponse) {}
+
+  // Update the state of a task with execution results. (also used in push
+  // models)
+  rpc SetTaskStatus(SetTaskStatusRequest) returns (SetTaskStatusResponse) {}
+}
+
+message GetTaskRequest {
+  optional string namespace = 1;
+}
+message GetTaskResponse {
+  // If there are no tasks available, these will be empty
+  optional TaskActivation task = 1;
+  optional google.protobuf.Timestamp processing_deadline = 2;
+
+  optional Error error = 3;
+}
+
+message SetTaskStatusRequest {
+  string id = 1;
+  string namespace = 2;
```

Comment:
Do we need `namespace` if we have the `id` of the activation?

Reply:
I included namespace here because I assumed workers would need to know which namespace they are updating, in order to make sure the write gets written back to the same consumer.

Reply:
Yeah OK I'll remove this. I think I just don't understand the load balancing model well enough yet.